eitprocessing/features/breath_detection.py

```diff
@@ -0,0 +1,425 @@
+import itertools
+import math
+from collections.abc import Callable
+from dataclasses import dataclass
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy import signal
+
+from eitprocessing.datahandling.breath import Breath
+from eitprocessing.datahandling.intervaldata import IntervalData
+from eitprocessing.datahandling.sequence import Sequence
+from eitprocessing.features.moving_average import MovingAverage
+
+
+@dataclass
+class BreathDetection:
+    """Algorithm for detecting breaths in data representing respiration.
+
+    This algorithm detects the position of breaths in data by detecting valleys
+    (local minimum values) and peaks (local maximum values) in data. When
+    initializing BreathDetection, the sample frequency of the data and the
+    minimum duration of a breath have to be provided. The minimum duration
+    should be short enough to include the shortest expected breath in the data.
+
+    Examples:
+    >>> bd = BreathDetection(sample_frequency=50, minimum_distance=0.5)
+    >>> breaths = bd.find_breaths(sequency=seq, continuousdata_label="global_impedance_(raw)")
+
+    Args:
+        sample_frequency: sample frequency of the data
+        minimum_distance: minimum expected distance between breaths, defaults to 0.67 seconds
+        averaging_window_length: length of window used for averaging the data, defaults to 15 seconds
```

Comment:
distance and length imply something different than duration. Also, please make the stated default exactly as the actual default (so either 2/3 in each case, or 0.67 in each case).